src/v/kafka/server/tests/consumer_groups_test.cc

```diff
@@ -140,6 +141,56 @@ FIXTURE_TEST(empty_offset_commit_request, consumer_offsets_fixture) {
     }
 }
 
+FIXTURE_TEST(block_test, consumer_offsets_fixture) {
+    model::topic t{"topic"};
+    kafka::group_id g{"group"};
+    kafka::group_instance_id gi("group-instance");
+
+    add_topic(model::topic_namespace_view{model::kafka_namespace, t}).get();
+    wait_for_consumer_offsets_topic(gi);
+
+    auto client = make_kafka_client().get();
+    auto deferred = ss::defer([&client] {
+        client.stop().then([&client] { client.shutdown(); }).get();
+    });
+    client.connect().get();
+
+    auto commit_offset = [&] {
+        auto req = offset_commit_request{.data{
+          .group_id = g,
+          .group_instance_id = gi,
+          .topics = chunked_vector<offset_commit_request_topic>::single(
+            offset_commit_request_topic{
+              .name = t,
+              .partitions
+              = chunked_vector<offset_commit_request_partition>::single(
+                offset_commit_request_partition{
+                  .partition_index = model::partition_id{0},
+                  .committed_offset = model::offset{0}})})}};
+        auto res = client.dispatch(std::move(req)).get();
+        return !res.data.errored();
+    };
+
+    auto set_blocked = [&](bool blocked) {
+        auto res = app._group_manager.local()
+                     .set_blocked_for_groups(
+                       model::ntp{
+                         model::kafka_namespace,
+                         model::kafka_consumer_offsets_topic,
+                         model::partition_id{0}},
+                       {g},
+                       blocked)
+                     .get();
+        BOOST_REQUIRE(res.has_value());
+    };
+
+    BOOST_REQUIRE(commit_offset());
+    set_blocked(true);
+    BOOST_REQUIRE(!commit_offset());
```

Comment:
leadership change to cover group_recovery_consumer code paths? (the new leader should also see the blocked group)